lib/rubocop/cop/rspec/class_check.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Enforces consistent use of `be_a` or `be_kind_of`.
+      #
+      # @example EnforcedStyle: be_a (default)
+      #   # bad
+      #   expect(var).to be_kind_of(Date)
+      #   expect(var).to be_kind_of(Integer)
+      #
+      #   # good
+      #   expect(var).to be_a(Date)
+      #   expect(var).to be_a(Integer)
+      #
+      # @example EnforcedStyle: be_kind_of
+      #   # bad
+      #   expect(var).to be_a(Time)
+      #   expect(var).to be_a(String)
+      #
+      #   # good
+      #   expect(var).to be_kind_of(Time)
+      #   expect(var).to be_kind_of(String)
+      #
+      class ClassCheck < Base
+        extend AutoCorrector
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<preferred>s` over `%<current>s`.'
+
+        RESTRICT_ON_SEND = %i[
+          be_a
+          be_kind_of
+        ].freeze
+
+        def on_send(node)
+          return if style == node.method_name
+
+          add_offense(
+            node.loc.selector,
+            message: format_message(node)
+          ) do |corrector|
+            autocorrect(corrector, node)
+          end
+        end
+
+        private
+
+        def autocorrect(corrector, node)
+          corrector.replace(
+            node.loc.selector,
+            opposite_method_name(node.method_name)
+          )
+        end
+
+        def format_message(node)
+          format(
+            MSG,
+            current: node.method_name,
+            preferred: opposite_method_name(node.method_name)
```

Comment:
How about the following?
```suggestion
            preferred: style
```